ast/desugar/Desugar.cc

```diff
@@ -1261,6 +1267,33 @@ TreePtr node2TreeImpl(DesugarContext dctx, unique_ptr<parser::Node> what) {
                     }
                 };
 
+                UnorderedSet<core::NameRef> hashKeySymbols;
+                auto &gs = dctx.ctx.state;
+                for (auto &key : keys) {
+                    auto lit = ast::cast_tree_const<ast::Literal>(key);
+                    core::NameRef nameRef;
+                    if (!lit) {
+                        continue;
+                    }
+                    if (lit->isSymbol(gs)) {
+                        nameRef = lit->asSymbol(gs);
+                    } else if (lit->isString(gs)) {
+                        nameRef = lit->asString(gs);
+                    } else {
+                        continue;
+                    }
+
+                    if (hashKeySymbols.find(nameRef) == hashKeySymbols.end()) {
+                        hashKeySymbols.insert(nameRef);
```

Comment:
What do you think about keeping track of a hash instead of a set, and map each key to it's first location (`core::LocOffset`)? Then you could use an `e.addErrorLine` to say where the previous key was.

A good test case here would be a large hash on multiple lines:

```ruby
{
  key: ...
  ...
  ...
  ...
  key: ...
}
```

Reply:
Nice idea! I've added cli test with that exact case.